test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -256,3 +265,95 @@
+   {:keys [aggregation breakout expressions fields filter limit]
+    :or   {expressions {"expr" convert-tz-expression}
+           filter      [:= [:field (mt/id :times :index) nil] 1]
+           fields      [[:expression "expr"]]}}]
+  (if breakout
+    (->> (mt/run-mbql-query times {:expressions expressions
+                                   :aggregation aggregation
+                                   :limit       limit
+                                   :filter      filter
+                                   :breakout    breakout})
+         mt/rows
+         ffirst)
+    (->> (mt/run-mbql-query times {:expressions expressions
+                                   :aggregation aggregation
+                                   :limit       limit
+                                   :filter      filter
+                                   :fields      fields})
+         mt/rows
+         ffirst)))
+
+(def offset->zone
+  "A map of all Offset to a zone-id.
+  {\"+07\00\" \"Asia/Saigon\"}"
+  (into {"+00:00" "UTC"}
+        (for [zone-id (java.time.ZoneId/getAvailableZoneIds)]
+         [(-> (t/zone-id zone-id)
+              .getRules
+              (.getOffset (java.time.Instant/now))
+              .toString)
+          zone-id])))
+
+(deftest convert-timezone-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :convert-timezone)
+    (mt/dataset times-mixed
+      (testing "timestamp with out timezone columns"
+        (testing "convert from +05:00 to +09:00"
+          (is (= "2004-03-19T13:19:09+09:00"
+                 (test-date-convert [:convert-timezone [:field (mt/id :times :dt) nil]
+                                                       (offset->zone "+09:00")
+                                                       (offset->zone "+05:00")]))))
+        (testing "convert to +09:00, from_tz should have default is system-tz (UTC)"
+          (is (= "2004-03-19T18:19:09+09:00"
+                 (test-date-convert [:convert-timezone [:field (mt/id :times :dt) nil] (offset->zone "+09:00")]))))
+
+        (mt/with-report-timezone-id "Europe/Rome"
+          (testing "from_tz should default to report_tz"
+            (is (= "2004-03-19T17:19:09+09:00"
+                   (test-date-convert [:convert-timezone [:field (mt/id :times :dt) nil] (offset->zone "+09:00")]))))
+
+          (testing "if from_tz is provided, ignore report_tz"
+            (is (= "2004-03-19T18:19:09+09:00"
+                   (test-date-convert [:convert-timezone [:field (mt/id :times :dt) nil]
+                                       (offset->zone "+09:00")
+                                       (offset->zone "+00:00")]))))))
+
+      (testing "timestamp with time zone columns"
+        (testing "convert to +09:00"
+          (is (= "2004-03-19T11:19:09+09:00"
+                 (test-date-convert [:convert-timezone [:field (mt/id :times :dt_tz) nil] (offset->zone "+09:00")]))))
+        (testing "timestamp with time zone columns shouldn't have `from_tz`"
```

Comment:
![Screen Shot 2022-10-05 at 22 42 57](https://user-images.githubusercontent.com/25661381/194103318-1f258f3d-7bb1-4a5b-b5d5-9b497d78b0bb.png)
You're right, that'd be the expected behavior. But my take is:
- Bruno pointed out we could ignore it(tho it's not his fav pick)
- Implementing this would require we do something equivalent to `u.date/with-time-zone-same-instant` but in SQL. I've tried to find a way but not sure how to do it properly. 

So that's why I went with throwing error.

But if you know how to do this in SQL, I'm all 👂s .

Reply:
I'm not quite sure what it means to ignore the time zone component, since that's not something that is stored. In postgres I can imagine using either
- `current_timestamp at time zone 'UTC'`, what the DB stores or
- `cast(current_timestamp as timestamp)`, what the DB shows if we hide the time zone offset.

Reply:
My interpretation of "ignore the timezone component" is:

Let's say we call `convertTimezone('2004-03-19 09:19:09+07:00', "+09:00", "+05:00")`

then should do:
1. ignore the original timezone, and pretend the column is created in `from_tz` instead. So `2004-03-19 09:19:09+07:00` => `2004-03-19 09:19:09+05:00`
2. then change timezone of this to "+09:00". So `2004-03-19 09:19:09+05:00` => `2004-03-19 13:19:09+09:00`

looping @brunobergher to double check.

Reply:
Yes, that's my thinking. Kind of force-casting the datetime to the other timezone, as if it were incorrect.

Reply:
That seems to be doable by a combination of casting to `timestamp` and then forcing the `from` timezone.

Reply:
I don't think so.
If we cast a `timestamp with time zone` column to `timestamp`, it shifts the hours back to `UTC`.

Then if you call `timezone('column', from-tz)` with it, this actually don't shift the hours to the `from-tz`, it's the same instant but with a tz.

```sql
select timezone('Asia/Ho_Chi_Minh', cast(timestamp with time zone '2004-03-19T07:19:09+07:00' as timestamp));
-- 2004-03-18 17:19:09+00

-- expect the results to be 2004-03-19T00:19:09+00:00
```